nixos/modules/security/pam.nix

```diff
@@ -1184,6 +1184,19 @@ in
       };
     };
 
+    security.pam.unixAuthArgs = mkOption {
+      default = "sha512";
+      type = types.str;
+      description = lib.mdDoc ''
+        Arguments to the pam_unix module.
+
+        Consider switching to "yescrypt rounds=10" for
+        better resistance against brute force, especially if
+        your login password is used as a key for encrypted
```

Comment:
I expect the most common case to be that the password is used to unwrap decryption keys _for the very root filesystem that contains the shadow file_. In that case choosing a weak hash function does not help an attacker who wishes to decrypt the filesystem, and yet this docstring suggests otherwise. I'm not sure how to phrase it in a way that is both succinct and precise, sadly.

Reply:
I believe you are talking of whole-disk-encryption. In that case, the decryption passphrase and login password are independent (and in general different, especially if there is more than one user).

ecryptfs & friends are generally used to encrypt user home directories, not /etc/shadow. For convenience the passphrase is encrypted using the login password of each user, so being able to brute force these is an issue.

Reply:
What happens when I just change this setting on my system?

Reply:
Would `types.listOf types.str` make more sense here?

Reply:
@SuperSandro2000, passwords in /etc/shadow will then be hashed using the methodology you selected.

Reply:
This affects the default for e.g. passwd.